support/cas-server-support-pac4j-core-clients/src/test/java/org/apereo/cas/support/pac4j/authentication/clients/DelegatedClientFactoryTests.java

```diff
@@ -0,0 +1,73 @@
+package org.apereo.cas.support.pac4j.authentication.clients;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import lombok.val;
+import org.apereo.cas.authentication.CasSSLContext;
+import org.apereo.cas.config.CasCoreHttpConfiguration;
+import org.apereo.cas.config.CasCoreUtilConfiguration;
+import org.apereo.cas.configuration.CasConfigurationProperties;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.pac4j.core.client.IndirectClient;
+import org.pac4j.core.profile.converter.AttributeConverter;
+import org.pac4j.saml.client.SAML2Client;
+import org.pac4j.saml.store.SAMLMessageStoreFactory;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.autoconfigure.RefreshAutoConfiguration;
+
+import javax.net.ssl.SSLContext;
+import java.util.Collection;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(classes = {
+        RefreshAutoConfiguration.class,
+        CasCoreUtilConfiguration.class,
+        CasCoreHttpConfiguration.class
+}, properties = {
+        "cas.authn.pac4j.saml[0].clientName=foo_client",
+        "cas.authn.pac4j.saml[0].identityProviderMetadataPath=file:/idp-metadata.xml",
+        "cas.authn.pac4j.saml[0].samlAttributeConverter=org.apereo.cas.support.pac4j.authentication.clients.DelegatedClientFactoryTests.FooConverter",
+})
+@EnableConfigurationProperties(CasConfigurationProperties.class)
+@Tag("SAML")
+public class DelegatedClientFactoryTests {
```

Comment:
Please remove this, and move tests to `DefaultDelegatedClientFactoryTests`.

Reply:
Done